src/locutus/model/terminology.py

```diff
@@ -44,12 +44,44 @@ def message(self):
 """
 
 
+class OntologyAPISearchPreference:
+    """
+    Represents a recorded Ontology API Search Preference
+    Attributes:
+        api_preference (dict): The users preference for ontology api and 
+        ontology.
+    """
+
+    def __init__(self, api_preference=None):
+        self.api_preference = api_preference or {}
+
+    class _Schema(Schema):
+        """
+        Marshmallow schema for serializing and deserializing Ontologies instances.
+
+        api_preference = {'api1': ['onto1', 'onto2'], 
+                          'api2': ['onto3']}
+        """    
+        api_preference = fields.Dict(keys=fields.Str(), values=fields.List(fields.Str()))
+
+        @post_load
+        def build_pref(self, data, **kwargs):
+            """
+            Builds an OntologyAPISearchPreference instance from deserialized data.
+            Args:
+                data (dict): The deserialized data.
+            Returns:
+                OntologyAPISearchPreference: An instance of the OntologyAPISearchPreference class.
+            """
+            return OntologyAPISearchPreference(**data)
+
 class Coding:
-    def __init__(self, code, display="", system=None, description=""):
+    def __init__(self, code, display="", system=None, description="", api_preference=None):

```

Comment:
I'm not really loving this. While this is a legitimate place to store these, I'm more inclined to see it as something that would live in it's own collection, similar to how we are storing mappings and provenance. The reason being that these codings are intended to represent something very basic, and these extensions are layers that are very specific to our application. If we stored information like this in the coding, we would need to filter it out before passing code systems and value sets to other systems (and the front end in certain places)